elide-core/src/main/java/com/yahoo/elide/security/permissions/ExpressionBuilder.java

```diff
@@ -200,106 +191,127 @@ public ExpressionBuilder(HashMap<Class<? extends Check>, Map<PersistentResource,
                 );
 
         return new Expressions(
-                buildAnyFieldExpression(resourceClass, entityDictionary, annotationClass, userCheckFn),
+                buildAnyFieldExpression(resourceClass, annotationClass, userCheckFn),
                 null
         );
     }
 
+    private Function<Check, Expression> getImmediateExpressionFor(PersistentResource resource, ChangeSpec changeSpec) {
+        return (check) -> new ImmediateCheckExpression(
+                check,
+                resource,
+                resource.getRequestScope(),
+                changeSpec,
+                cache
+        );
+    }
+
+    private Function<Check, Expression> getDeferredExpressionFor(PersistentResource resource, ChangeSpec changeSpec) {
+        return (check) -> new DeferredCheckExpression(
+                check,
+                resource,
+                resource.getRequestScope(),
+                changeSpec,
+                cache
+        );
+    }
+
+    private Expression expressionFromDefinition(PermissionDefinition permissions, Function<Check, Expression> checkFn) {
+        if (permissions == null) {
+            return null;
+        }
+
+        final BiFunction<Expression, Expression, Expression> expressionJoiner = getJoiner(permissions.permissionType);
+        switch (permissions.permissionType) {
+            case EXPRESSION:
+                ExpressionVisitor expressionBuilder = new ExpressionVisitor(entityDictionary, checkFn);
+                return expressionBuilder.visit(permissions.expressionTree);
+
+            case ANY:
```

Comment:
See comment above about `PermissionDefinition`. It would be nice to eliminate this code if possible.